### PR TITLE
Only in watch mode option

### PR DESCRIPTION
**Pull request by andykais**
based off of #4 
- add option `onlyInWatchMode`, default to true
  - otherwise, clear if neither `mode` nor `NODE_ENV` equal `"production"`
- update readme

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
-        if (process.env.NODE_ENV !== "production") this.clearConsole();
+      compiler.hooks.afterCompile.tap('CleanTerminalPlugin', () => {
+        if (this.shouldClearConsole(compiler)) this.clearConsole();
       });
     } else {
       // backwards compatible version of compiler.hooks
-      compiler.plugin("emit", (_, done) => {
-        if (process.env.NODE_ENV !== "production") this.clearConsole();
+      compiler.plugin('emit', (_, done) => {
+        if (this.shouldClearConsole()) this.clearConsole();
```
Comment (mrnocreativity) [inline index.js]: There's an error here (which is currently still in in the NPM package). 
The compiler is not passed to the shouldClearConsole method which triggers an undefined-error ;)

Comment (danillouz) [reply index.js]: Thanks for the heads up, created #12

Comment (mrnocreativity) [reply index.js]: I added a PR that fixed the issue :)

Comment (andykais) [reply index.js]: Thanks for the catch @mrnocreativity. Apologies guys, I should've spent more time looking this over before pr-ing 😞

Comment (mrnocreativity) [reply index.js]: No worries :) That's what community is for!

Comment (danillouz) [reply index.js]: @andykais no worries, I also missed it while reviewing :)
@mrnocreativity Thanks for the PR, I wasn't sure you'd be sending one, so I created an issue as well.

Moving the conversation to #11

**Pull request merged**